File: backend/types/payload/record.go
package payload

import "chouxcream-backend/types/enum"

type Record struct {
	Id           *uint64    `json:"id"`
	MenuId       *uint64    `json:"menu_id" validate:"omitempty,gt=0"`
	MenuName     *string    `json:"menu_name"`
	MenuImageUrl *string    `json:"menu_image_url"`
	Meal         *enum.Meal `json:"meal"`
	Calorie      *float64   `json:"calorie" validate:"gte=0"`
	Protein      *float64   `json:"protein" validate:"gte=0"`
	Carbohydrate *float64   `json:"carbohydrate" validate:"gte=0"`
	Fat          *float64   `json:"fat" validate:"gte=0"`
}

File: app/lib/models/menu/record.dart
import 'package:json_annotation/json_annotation.dart';

part 'record.g.dart';

@JsonSerializable()
class Record {
  @JsonKey(name: 'id')
  final int id;

  @JsonKey(name: 'menu_name')
  final String menuName;

  @JsonKey(name: 'menu_image_url')
  final String menuImageUrl;

  @JsonKey(name: 'meal')
  final String meal;

  @JsonKey(name: 'calorie')
  final double calorie;

  @JsonKey(name: 'protein')
  final double protein;

  @JsonKey(name: 'carbohydrate')
  final double carbohydrate;

  @JsonKey(name: 'fat')
  final double fat;

  Record(
    this.menuName,
    this.menuImageUrl,
    this.meal,
    this.calorie,
    this.protein,
    this.carbohydrate,
    this.fat, {
    required this.id,
  });

  factory Record.fromJson(Map<String, dynamic> json) => _$RecordFromJson(json);

  Map<String, dynamic> toJson() => _$RecordToJson(this);
}
